Aptoide/aptoide-client-v8
# [DT-336] My Profile

**What does this PR do?**

This PR aims at implementing the MyProfile View Page that can be accessed through the Action top bar.
It allows the user to edit his username and profile picture and access his settings, as well as Logout, delete account and navigate through Aptoide Products.
These features were implemented using Android DataStore to store and edit user Preferences and ActivityResultContracts for permissions and access to Camera and gallery files.

**Database changed?**

   Yes 

**Where should the reviewer start?**

- [ ] MainActivity.kt
- [ ] BottomNavigationView.kt
- [ ] MyProfileView.kt
- [ ] EditProfileView.kt
- [ ] MyFileProvider.kt
- [ ] file_paths.xml
- [ ] settingsView.kt
- [ ] sendFeedbackView.kt
- [ ] urlViewScreen.kt
- [ ] Utils.kt
- [ ] RepositoryModule.kt
- [ ] UserProfileViewModel.kt
- [ ] UserProfileUseCase.kt
- [ ] UserProfileManager.kt
- [ ] UserPreferencesViewModel.kt
- [ ] UserPreferencesUseCase.kt
- [ ] UserPreferencesManager.kt
- [ ] ManagedKeys.kt
- [ ] DeviceInfoViewModel.kt
- [ ] DeviceInfoRepository.kt
- [ ] AptoideApplication.kt
- [ ] Analytics.kt
- [ ] AnalyticsViewModel.kt
- [ ] ThemeListener.kt
- [ ] Topbar.kt
- [ ] AppIcons.kt
- [ ] AppColors.kt
- [ ] Gradients.kt
- [ ] Color.kt

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-336](https://aptoide.atlassian.net/browse/DT-336)
  Another way to test this is, through the action top bar, click on user icon and it should take you to MyProfileView, it's possible to edit the username and user image by click on the user icon or the "edit" text in right upper corner, and settings View available in the Card Button. Everything should work just fine

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-336](https://aptoide.atlassian.net/browse/DT-336)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**


[DT-336]: https://aptoide.atlassian.net/browse/DT-336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
